lib/remoteResource.js

```diff
@@ -26,16 +26,71 @@ const requestsTemplate = `{
   }
 }`;
 
+const requestsTemplateGit = `{
+  "options": {
+    "git": {
+      "provider": "{{{provider}}}",
+      "repo": "{{{repo}}}",
+      "branch": "{{{branch}}}",
+      "filePath": "{{{filePath}}}"
+    },
+    "headers": {
+      "razee-org-key": {
+        "valueFrom": {
+          "secretKeyRef":{
+            "name": "{{{secretName}}}",
+            "namespace": "{{namespace}}",
+            "key": "razee-api-org-key"
+          }
+        }
+      },
+      "Authorization": {
+        "valueFrom": {
+          "secretKeyRef":{
+            "name": "{{{secretNameGit}}}",
+            "namespace": "{{namespace}}",
+            "key": "token"
+          }
+        }
+      }
+    }
+    
+  }
+}`; 
+
 const createRemoteResources = async (razeeApi, apiKey, subscriptions, clusterId) => {
   log.info('create remote resources subscription list', { subscriptions });
   try {
     return Promise.all(subscriptions.map(async sub => {
       const apiKeyBase64 = Buffer.from(apiKey).toString('base64');
       const url = `${razeeApi}/${sub.url}`;
       const secretName = `clustersubscription-${sub.subscriptionUuid}-secret`;
-      const rendered = Mustache.render(requestsTemplate, { url: url, secretName: secretName, namespace: NAMESPACE });
+      let rendered;
+      let backendService;
+      let secretNameGit = 'satellite-config-gitops';
+      if (sub.url) {
+        rendered = Mustache.render(requestsTemplate, { url: url, secretName: secretName, namespace: NAMESPACE });
+        backendService = 'generic';
+      } else if (sub.remote) {
+        const provider = sub.remote.remoteType;
```

Comment:
Does the remoteType value need to be checked to differentiate between `github` and `gitlab` or an unexpected value?  Similarly, does the `backendService = 'git'` need to differentiate between `github` and `gitlab` or other?

Reply:
backendservice doesn't need to differentiate, rr will handle it